### PR TITLE
feat: New Header styles for Beam tables

### PR DESCRIPTION
**Pull request by bdow**
Creates new GridTable story for showing new header styles

### CODE REVIEW

File: src/components/Table/GridTable.stories.tsx [hunk 1]
```diff
@@ -724,6 +730,99 @@ export function CustomEmptyCell() {
...
+        </div>
+      ),
+      sortValue: tradeCategories.sort().join(" "),
+    }),
+  });
+  const dateCol = dateColumn<BeamRow>({ header: "Date", data: ({ date }) => date });
+  const locationCol = column<BeamRow>({
+    header: "Location",
+    data: (row) => ({
+      content: <SelectField value={row.location} onSelect={noop} options={locations} label="Location" hideLabel />,
```
Comment (stephenh) [inline src/components/Table/GridTable.stories.tsx:785]: Nudge for GridTable to use a PresentationProvider to auto-`hideLabel` all of these :-)

**Review by stephenh — APPROVED** [review]

File: src/components/Table/GridTable.stories.tsx (hunk 1 above)
Comment (bdow) [reply src/components/Table/GridTable.stories.tsx:785]: Great point, this is already done... I forgot!

Comment (stephenh) [reply src/components/Table/GridTable.stories.tsx:785]: Hot damn! :-D

**Comment by bdow** [comment]
@stephenh - after @jeffreyqua went through with QA it was determined design wants the sort icon to show up on the left of the text for right aligned columns. I've updated the PR in this commit to support that: https://github.com/homebound-team/beam/pull/432/commits/95dd70d9e3e69d1c252075d6d9b392c5989146cd
Would you mind re-reviewing. Would like to get your blessing on this approach.

**Comment by stephenh** [comment]
@bdow thanks for the ping! Yep, that looks great!

**Review by stephenh — APPROVED** [review]

**Pull request merged**